src/Services/OpenApiGenerator.php

```diff
@@ -205,17 +214,31 @@ private function getBodyParametersFromMethodSignature(Route $route): array
             return [];
         }
 
-        if (null === $req = $this->getFromRequestFromMethodArguments($route)) {
+        if (null === $req = $this->getFormRequestFromMethodArguments($route)) {
             return [];
         }
 
-        $rules    = implode(' ', array_map(fn ($v) => is_array($v) ? implode(' ', $v) : $v, array_values($req->rules())));
-        $required = array_keys(array_filter($req->rules(), fn ($v, $k) => !str_contains($k, '.') && str_contains(is_array($v) ? implode(' ', $v) : $v, 'required'), ARRAY_FILTER_USE_BOTH));
+        $normalized = [];
+        foreach ($req->rules() as $prop => $spec) {
+            $normalized[$prop] = $this->normalizeRuleSpec($spec);
+        }
+        $bigRulesString = implode(' ', $normalized);
+
+        $required = [];
+        foreach ($normalized as $prop => $spec) {
```

Comment:
This treats the `field.*.field` as being the path itself, so it will add e.g.: `permissions.*.id` to the list of required fields, but this is not a valid field name. Those fields should be defined on the child object and not in the parent. For example the resulting (abbreviated) generated code segment from this update is:

```yaml
type: object
    required:
        - account_id
        - email
        - password
        - name
        - roles.*.id
        - permissions.*.id
    properties:
        roles:
            type: array
            items:
                type: object
                properties:
                    id:
                        type: string
```
but it should result in (edited for brevity):
```yaml
type: object
    required:
        - account_id
        - email
        - password
        - name
    properties:
        roles:
            type: array
            items:
                type: object
                required:
                    - id
                properties:
                    id:
                        type: string
```